dlib/dnn/tensor_tools.h

```diff
@@ -947,6 +947,24 @@ namespace dlib { namespace tt
                   this gradient to filters_gradient.
         !*/
 
+        void operator() (
+            resizable_tensor& output,
+            const tensor& data,
+            const tensor& filters
+       )
+       {
+          impl.forward_without_setup(output,data,filters);
+       }
+
+       void setup(
```

Comment:
Thanks

Yes I agree with your statement on interface complexity, my feeling was I didn't want to change too much of the original code (the setup code was already there but didn't fit well when the order of things are swapped). I am rather conservative :). 

But changing the signature of the setup code to include the right set of parameters instead of relying to the current pre-assumptions sounds like a good compromise I will have a look later and see if I can adjust it.